src/components/implementation/cbuf_mid/mid_level/cbuf_mid.c

```diff
@@ -38,8 +38,8 @@ void cbuf_call(char c)
 	for (i = 0; i < iter ; i++){
 		int tmem;
 		cbt[i] = cbuf_null();
-		mt[i] = cbuf_alloc(SZ, &cbt[i]);
-		cbuf_unpack(cbt[i], &id, &tmem);
+		mt[i] = cbuf_alloc(SZ, &cbt[i], 1);
```

Comment:
1. The third argument shouldn't be a boolean.  It should be a flag, so that we can expand it later if needs be.

2. We probably don't want the third argument to be part of the common-case API.  TMem will mostly only be used by the IDL, so we should just provide another function that enables us to pass in the flags.  Perhaps something like `void *cbuf_alloc_ext(unsigned int sz, cbuf_t *cbid, unsigned int flags)`.  Most code that explicitly calls cbuf functions will not be using the flags (unless one of them in the future becomes essential), so we might as well optimize for that common case, and have `cbuf_alloc(sz, *cb)` be the function prototype.  This will obviously be a simple wrapper around the `_ext` function.

What do you guys thing?  Is it worth optimizing away the flags as we will essentially never use them in common code?